.github/workflows/linux-lint.yml

```diff
@@ -0,0 +1,40 @@
+name: Linux Lint
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install system dependencies
+      run: |
+        # build-essential libboost-all-dev clang-9 ruby-full cmake
+        sudo apt-get install libc++-dev libc++abi-dev ninja-build lcov llvm cppcheck
+        cppcheck --version
+    - name: Cache internal dependencies
+      id: internal_cache
+      uses: actions/cache@v1.1.2
+      with:
+        path: build/local
+        key: ${{ runner.os }}-internal-${{ hashFiles('tools/install-dependencies') }}
+    - name: Install internal dependencies
+      run: |
+        tools/install-dependencies
+      env:
+        CC: /usr/bin/clang
+        CXX: /usr/bin/clang++
+      if: steps.internal_cache.outputs.cache-hit != 'true'
```

Comment:
would this lint be consistent with local?

Reply:
All linting settings are consistent, coming from Cmake and cppcheck exception files.  What could be different is the cppcheck version -- this build installs with apt-get on Ubuntu 18.04, version is 1.82, which is not the latest.